x-pack/plugins/monitoring/server/kibana_monitoring/lib/type_collector.js

```diff
@@ -47,15 +52,13 @@ export class TypeCollector {
   }
 
   /*
-   * @param {Object} collectorOptions (required)
-   * Fields:
-   * - type {String}
-   * - init {Function} (optional)
-   * - fetch {Function}
-   * - cleanup {Function} (optional)
+   * @param {String} type.type
+   * @param {Function} type.init (optional)
+   * @param {Function} type.fetch
+   * @param {Function} type.cleanup (optional)
    */
-  register(collectorOptions) {
-    this._collectors.push(collectorOptions);
+  register(type) {
```

Comment:
One downside to this approach is that a plugin that registers late could technically miss [at least] the first collection cycle.

I don't have a better approach in mind, but it's something to keep in mind as we add more.